xonsh/aliases.py

```diff
@@ -104,13 +104,12 @@ def bang_bang(args, stdin=None):
 def bash_aliases():
     """Computes a dictionary of aliases based on Bash's aliases."""
     try:
-        s = subprocess.check_output(['bash', '-i', '-l'],
-                                    input='alias',
+        s = subprocess.check_output(['bash', '-i', '-c', 'alias'],
                                     stderr=subprocess.PIPE,
                                     universal_newlines=True)
     except (subprocess.CalledProcessError, FileNotFoundError):
         s = ''
-    items = [line.split('=', 1) for line in s.splitlines() if '=' in line]
+    items = [line.split('=', 1) for line in s.splitlines() if line.startswith('alias ') and '=' in line]
```

Comment:
To be fair, this project would probably cause flake8 to light everything up like a Christmas tree, if I hadn't ignored some things. :stuck_out_tongue_winking_eye: